### PR TITLE
Fix flaky uao_crash_compaction_column test

### PR DESCRIPTION
**Pull request by pengzhout**
In a normal vacuum, for each column, the vacuum will
1) create a new segment file and record it in the gp_aocsseg_xxxxx
   table with a new entry with frozen xmin.
2) compact the tuples in the old segment file and move it to the new
   segment file.
3) drop/truncate the old segment file and mark it as dropped in the
   gp_aocsseg_xxxxx table.

In the uao_crash_compaction_column test, a PANIC is injected between
step 1 and step2 on seg0, so the whole transaction is aborted.

For seg1 and seg2, because step1 is using a frozen xmin, the new segment
file is visible even the vacuum is aborted.

For seg0, things are different. Because the PANIC occurs right behind
the new segment file is created in a very short time, so the buffer cache
for the new record in the gp_aocsseg_xxxx and the buffer cache for the
related xlog have a big chance to not be flushed to the disk, so when the
seg0 is recovered, we will not see the record in the gp_aocsseg_xxxx. But
if the buffer is flushed to the disk because the checkpointer/bgwriter
before the panic, we will see the new segment file record in the
gp_aocsseg_xxxx table.

This is RCA of the flakiness.

Considering the purpose of the test case, this commit separates the query
to two queries:

1. "SELECT segno,column_num,physical_segno,tupcount,modcount,state FROM
   gp_toolkit.__gp_aocsseg('crash_vacuum_in_appendonly_insert') where
   segno = 1" to verify the old segment is still visible after the vacuum
   is aborted.
2. "SELECT sum(tupcount) FROM gp_toolkit.__gp_aocsseg('crash_vacuum_in_ap
   pendonly_insert') where state = 1 and segno = 2" to verify the new
   segment file contains no tuples.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by gfphoenix78** [comment]
After the PANIC during the vacuum, the new segment file(segno=2) created on the seg0 causes the flaky behavior. Is there any way to avoid creating the segment file(segno=2) on the seg0?

File: src/test/isolation2/expected/uao_crash_compaction_column.out [hunk 1]
```diff
@@ -232,13 +233,13 @@ UPDATE 1
...
- 2     | 0          | 2              | 0        | 0        | 1     
- 2     | 0          | 2              | 0        | 0        | 1     
- 2     | 1          | 130            | 0        | 0        | 1     
- 2     | 1          | 130            | 0        | 0        | 1     
- 2     | 2          | 258            | 0        | 0        | 1     
- 2     | 2          | 258            | 0        | 0        | 1     
-(15 rows)
+(9 rows)
+-- verify the new segment files contain no tuples.
+1:SELECT sum(tupcount) FROM gp_toolkit.__gp_aocsseg('crash_vacuum_in_appendonly_insert') where state = 1 and segno = 2;
```
[the remark this answers is not in the document]
Comment (pengzhout) [reply src/test/isolation2/expected/uao_crash_compaction_column.out:238]: seems unnecessary actually, thanks, will remove it.

**Comment by pengzhout** [comment]
> After the PANIC during the vacuum, the new segment file(segno=2) created on the seg0 causes the flaky behavior. Is there any way to avoid creating the segment file(segno=2) on the seg0, or at least remove the entry from pg_aocsseg_xxx?

Hmm, the fault is injected after the new segment file is created with a frozen xmin, so we can 1) move the fault ahead or 2) don't use a frozen xmin, neither seem to be a good option.

**Comment by gfphoenix78** [comment]
@pengzhout  I saw that the `vacuum aocs_table` is not a single-transaction query. It has more than one transaction(See that from the xmin/xmax in pg_aoseg.pg_aocsseg_xxx). So do we know which transaction the PANIC interrupts? If the interrupted transaction is undetermined, we'd better update the test/inject point to control better.

**Comment by pengzhout** [comment]
Thanks for the review, merged.

**Pull request merged**